mod/state-transition/pkg/core/state_processor.go

```diff
@@ -81,10 +81,18 @@ type StateProcessor[
 	cs common.ChainSpec
 	// signer is the BLS signer used for cryptographic operations.
 	signer crypto.BLSSigner
+	// fGetAddressFromPubKey verifies that a validator public key
+	// matches with the proposer address passed by the consensus
+	// Injected via ctor to simplify testing.
+	fGetAddressFromPubKey func(crypto.BLSPubkey) ([]byte, error)
```

Comment:
`fGetAddressFromPubKey` avoids the inconvenience of setting BLS keys in unit tests, since it allows injecting whatever function we care to pass or not validation.
It is useful as we merge in unit tests and we need to pass proposer address checks